ably/auth.go

```diff
@@ -42,21 +63,87 @@ func (c Capability) String() string {
 }
 
 type Token struct {
```

Comment:
@mattheworiordan I already got that covered - I've reworked whole Auth [on a separate branch](https://github.com/rjeczalik/ably-go/commit/2f76de11810e7710483a4dd6673a2fd43e0e1b4d#diff-fe17adcb123ff94418b17414c9d8940dR111), the PR will follow after this one.